### PR TITLE
TVG Loader/Saver module: tvg helper file

### PR DESCRIPTION
**Pull request by mmaciola**
tvgTvgHelper.h is a base file used for .tvg saver/loader module

### CODE REVIEW

File: src/lib/tvgTvgHelper.h [hunk 1]
```diff
@@ -0,0 +1,82 @@
+#ifndef _TVG_STANDARD_HELPER_H_
```
Comment (JSUYA) [inline src/lib/tvgTvgHelper.h:1]: Is there any reason this Helper header file should be in src/lib?
I think it should be in src/loaders/tvg if it's only used in tvgTvgLoader.
And if you move this header file, I think proper name of this file is tvgTvgLoaderHelper.h.

File: src/lib/tvgTvgHelper.h [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+#define _TVG_STANDARD_HELPER_H_
+
+#include "tvgCommon.h"
```
Comment (JSUYA) [inline src/lib/tvgTvgHelper.h:4]: Do we need this?

**Review by JSUYA — CHANGES_REQUESTED** [review]
Please check my comment

File: src/lib/tvgTvgHelper.h (hunk 1 above)
Comment (mgrudzinska) [reply src/lib/tvgTvgHelper.h:1]: it is also used by the saver, but I need a moment to decide where the saver will be implemented

File: src/lib/tvgTvgHelper.h [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+#define _read_tvg_float(dst, src) memcpy(dst, (src), sizeof(float))
+
+using TvgIndicator = uint8_t;
+using ByteCounter = uint32_t;
+using TvgFlag = uint8_t;
+#define TVG_INDICATOR_SIZE sizeof(TvgIndicator)
+#define BYTE_COUNTER_SIZE sizeof(ByteCounter)
+#define TVG_FLAG_SIZE sizeof(TvgFlag)
+
+struct tvg_block
```
[the remark this answers is not in the document]
Comment (mmaciola) [reply src/lib/tvgTvgHelper.h]: renamed to tvgBlock

File: src/lib/tvgTvgHelper.h (hunk 2 above)
Comment (mmaciola) [reply src/lib/tvgTvgHelper.h]: removed include line

File: src/lib/tvgTvgHelper.h (hunk 1 above)
Comment (mmaciola) [reply src/lib/tvgTvgHelper.h]: Renamed `_TVG_STANDARD_HELPER_H_` into `_TVG_TVG_HELPER_H_` and added copyright header

**Review by JSUYA — APPROVED** [review]
I think this PR is appropriate.
When the review of other patches is complete, I will merge this.

**Review by hermet — CHANGES_REQUESTED** [review]
@mmaciola Looks bad that this file make a coupling between tvg loader & common lib.
In point of tvg common lib view, It's not necessary. How about moving the loader side?
Also, commit message is too weak. at least you can mention any rules&policy to maintain these macro values.

**Comment by hermet** [comment]
@mmaciola + question, is that right whenever new property is added, we should update the macro and loader code?

**Comment by mmaciola** [comment]
This file will be used by saver module too; it must be accessible for both loader and saver.

When new features added, we will just add new definitions into this file. Existing definitions won't be changed. Loader and saver modules will be then changed accordingly to handle new features.

I changed commit message. Please write what you think

**Review by hermet — APPROVED** [review]

**Pull request merged**